pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -199,32 +227,274 @@ pub mod pallet {
 			todo!()
 		}
 
-		// TODO: not sure if we really need that. Check that out with Raphael.
 		#[pallet::call_index(6)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn start_destroy(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn start_refund(origin: OriginFor<T>, pool_id: T::PoolId, currency_count: u32) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			Self::do_start_refund(pool_id, Some(&who), currency_count)?;
+
+			Ok(())
 		}
 
 		#[pallet::call_index(7)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn force_start_destroy(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn force_start_refund(origin: OriginFor<T>, pool_id: T::PoolId, currency_count: u32) -> DispatchResult {
+			T::ForceOrigin::ensure_origin(origin)?;
+
+			Self::do_start_refund(pool_id, None, currency_count)?;
+
+			Ok(())
 		}
 
-		// todo: check if we really need that tx.
 		#[pallet::call_index(8)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn destroy_accounts(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn refund_account(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			account: AccountIdLookupOf<T>,
```

Comment:
Is it not the same as `AccountIdOf<T>`?

Reply:
With a lookup you have to lookup first (line 260). This  helps using any account lookup functionality that the chain may have (e.g., by index)